### PR TITLE
add unit tests for module_utils/resource.py

### PR DESCRIPTION
**Pull request by abikouo**
add unit tests for `module_utils/resource.py`

### CODE REVIEW

File: plugins/module_utils/resource.py [hunk 1]
```diff
@@ -28,7 +28,10 @@
 def get_value(data, path):
     while path:
         key = path.pop(0)
-        data = data[key]
+        if isinstance(data, dict) and key in data:
+            data = data[key]
+        else:
+            raise KeyError(f"{data} is not of type dict or does not contain key {key}")
```
Comment (gravesm) [inline plugins/module_utils/resource.py:34]: Why hide a TypeError behind a KeyError? The KeyError is expected in the case of check mode, but if we are getting a TypeError here, I think that would indicate a bigger problem.

File: tests/unit/module_utils/test_resource.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+@patch(PATCH_BASE_PATH + "replace_reference")
+def test_resolve_refs_with_check_mode(m_replace_reference):
+
+    context = {}
+
+    def raise_key_error():
+        return KeyError("missing key from dictionnary")
+
+    m_replace_reference.side_effect = raise_key_error()
```
Comment (gravesm) [inline tests/unit/module_utils/test_resource.py:98]: Can't you just do `m_replace_reference.side_effect = KeyError('some key')`?

File: tests/unit/module_utils/test_resource.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+    sorter = cloudclient.sort_resources(desired_state, state)
+    result = []
+    while sorter:
+        for name in sorter.get_ready():
+            result.append(name)
+            sorter.done(name)
+    if state == "absent":
+        expected.reverse()
+    print(f"Expected: {expected} - Result: {result}")
```
Comment (gravesm) [inline tests/unit/module_utils/test_resource.py:155]: Is this left over from debugging? Can it be removed?

Comment (abikouo) [reply tests/unit/module_utils/test_resource.py:155]: will be removed

File: tests/unit/module_utils/test_resource.py (hunk 2 above)
Comment (abikouo) [reply tests/unit/module_utils/test_resource.py:98]: done

**Review by gravesm — APPROVED** [review]

**Pull request merged**